src/Orleans/Utils/IKeyedServiceCollection.cs

```diff
@@ -0,0 +1,37 @@
+﻿
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Collection of services that can be disambiguated by key
+    /// </summary>
+    public interface IKeyedServiceCollection<TKey, TService>
+        where TService : class
+    {
+        TService GetService(TKey key);
+    }
+
+    public static class KeyedServiceExtensions
+    {
+        /// <summary>
+        /// Acquire a service by key.
+        /// </summary>
+        public static TService GetServiceByKey<TKey, TService>(this IServiceProvider services, TKey key)
```

Comment:
I'm good with that.  I'm dyslexic so I don't trust my ordering judgment.  What is intuitive to me rarely works for others.

For instance, I think ok button should be on the right and cancel should be on the left, like GitHub does it, which is wrong by most UI design rules. :)